src/services/wallet-service/wallet-service.spec.ts

```diff
@@ -0,0 +1,177 @@
+    let mockedWalletService: MockWalletService;
+
+    beforeEach(() => {
+        storage = new MockStorage();
+        merkleTreeService = new MockMerkleTreeService();
+        mockedWalletService = new MockWalletService();
+        walletService = new WalletService(storage, <MerkleTreeService>merkleTreeService);
+    });
+
+    it("should return an empty array when there are no wallets saved", (done) => {
+        spyOn(storage, "get").and.returnValue(Promise.resolve());
+        walletService.getAll().then(data => {
+            expect(storage.get).toHaveBeenCalledWith(WALLET_STORAGE_KEY);
+            expect(data).toEqual([]);
+            done();
+        });
+    });
+
+    it("should return three specific wallets when retrieving the wallets from storage", (done) => {
+        mockedWalletService.getAll().then(data => {
+            spyOn(storage, "get").and.returnValue(Promise.resolve(data));
+            walletService.getAll().then(data => {
+                expect(data).toEqual(<any>[
+                    {"id":"012d294e-cb11-439b-937a-12d47a52c305",
+                        "type":"local",
+                        "name":"Biosta",
+                        "publicKey":
+                        "ETm9QUJLVdJkTqRojTNqswmeAQGaofojJJ",
+                        "encryptedPrivateKey":"E9873D79C6D87DC0FB6A5778633389F4453213303DA61F20BD67FC233AA33262"},
+                    {"id":"9b5329ff-c683-42a5-9165-4093e4076166",
+                        "type":"local",
+                        "name":"Labilo",
+                        "publicKey":"ELsKCchf9rcGsufjRR62PG5Fn5dFinfgeN",
+                        "encryptedPrivateKey":"E9873D79C6D87DC0FB6A5778633389F4453213303DA61F20BD67FC233AA33262"},
+                    {"id":"a2e16167-fedb-47d2-8856-2b3f97389c35",
+                        "type":"local",
+                        "name":"Zalista",
+                        "publicKey":"EZ7tP3CBdBKrB9MaBgZNHyDcTg5TFRRpaY",
+                        "encryptedPrivateKey":"E9873D79C6D87DC0FB6A5778633389F4453213303DA61F20BD67FC233AA33262"}]);
+                done();
+            });
+        });
+    });
+
+    it("should return cloned wallets when there are already existing wallets", () => {
+        spyOn(storage, "get").and.callThrough();
+        (<any>walletService).wallets = [
+            {"id":"012d294e-cb11-439b-937a-12d47a52c305",
+                "type":"local",
+                "name":"Biosta",
+                "publicKey":
+                "ETm9QUJLVdJkTqRojTNqswmeAQGaofojJJ",
+                "encryptedPrivateKey":"E9873D79C6D87DC0FB6A5778633389F4453213303DA61F20BD67FC233AA33262"}];
+        walletService.getAll();
+        expect(storage.get).not.toHaveBeenCalled();
+    });
+
+    it("should push a new wallet when it does not yet exist and overwrite when it already exists", (done) => {
+        // So that getWalletIndex doesn't crash on undefined length
+        (<any>walletService).wallets = [];
+        // Let to reset later
```

Comment:
What do you mean with this comment?

Reply:
At start, wallets array is undefined and therefor I cannot set a spy on the push method. By setting it to an empty array I can spy on the push method

Reply:
But.. I think setting the wallets array to a defined empty one and checking in getAll if the length is greater than 0 instead of checking if defined is a better solution